File: Data Structures & Algorithms/two-integer-sum/submission-4.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let len=nums.len();
        let mut num_pos: HashMap::<i32, i32>=HashMap::new();
        for (i, &num) in nums.iter().enumerate(){
            let complement=target-num;
            if let Some(&pos)=num_pos.get(&complement){
                return vec![pos, i as i32];
            }
            num_pos.insert(num, i as i32);
        }
        return vec![];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l=0;
        int r=numbers.length-1;
        while(l<r){
            while(l<r && numbers[l]+numbers[r]<target) l++;
            while(l<r && numbers[l]+numbers[r]>target) r--;
            if(numbers[l]+numbers[r]==target) return new int[]{l+1, r+1};
        }
        return new int[2];
    }
}
